src/app/reducers/trips.reducer.ts

```diff
@@ -6,69 +7,74 @@ import { Trip } from './../models/trip';
 import { combineLatest } from 'rxjs/observable/combineLatest';
 
 export interface State {
-    ids: string[];
-    trips: { [id: string]: Trip };
-    selectedTripId: string;
-    selectedCityId: string;
+	trips: TripsState;
+	selectedTripId: string;
+	selectedCityId: string;
 }
 
 const initialState = {
-    ids: [],
-    trips: {},
-    selectedTripId: null,
-    selectedCityId: null
+	trips: {
+		ids: [],
+		trips: {},
```

Comment:
Why ids and trips ? I think we are using a wrong data structure, should be something like `{id: trip}` correct me if I am missing something.